container/container_test.go

```diff
@@ -0,0 +1,124 @@
+	"time"
+
+	"github.com/stvp/assert"
+)
+
+func TestFindContainerNotExisting(t *testing.T) {
+	namespace := []string{"TestFindContainerNotExisting"}
+	container, err := FindContainer(namespace)
+	assert.Nil(t, err)
+	assert.Nil(t, container)
+}
+
+func TestFindContainerStrictNotExisting(t *testing.T) {
+	namespace := []string{"TestFindContainerStrictNotExisting"}
+	container, err := FindContainerStrict(namespace)
+	assert.NotNil(t, err)
+	assert.Nil(t, container)
+}
+
+func TestFindContainer(t *testing.T) {
+	namespace := []string{"TestFindContainer"}
+	startTestService(namespace)
+	defer StopService(namespace)
+	<-WaitContainerStatus(namespace, RUNNING, time.Minute)
+	container, err := FindContainer(namespace)
+	assert.Nil(t, err)
+	assert.NotNil(t, container)
+}
+
+func TestFindContainerStopped(t *testing.T) {
+	namespace := []string{"TestFindContainerStopped"}
+	startTestService(namespace)
+	StopService(namespace)
+	container, err := FindContainer(namespace)
+	assert.Nil(t, err)
+	assert.Nil(t, container)
+}
+
+func TestContainerStatusNeverStarted(t *testing.T) {
+	namespace := []string{"TestContainerStatusNeverStarted"}
+	status, err := ContainerStatus(namespace)
+	assert.Nil(t, err)
+	assert.Equal(t, status, STOPPED)
+}
+
+func TestContainerStatusRunning(t *testing.T) {
+	namespace := []string{"TestContainerStatusRunning"}
+	startTestService(namespace)
+	defer StopService(namespace)
+	<-WaitContainerStatus(namespace, RUNNING, time.Minute)
+	status, err := ContainerStatus(namespace)
+	assert.Nil(t, err)
+	assert.Equal(t, status, RUNNING)
+}
+
+func TestContainerStatusStopped(t *testing.T) {
+	namespace := []string{"TestContainerStatusStopped"}
+	startTestService(namespace)
+	<-WaitContainerStatus(namespace, RUNNING, time.Minute)
+
+	StopService(namespace)
+	<-WaitContainerStatus(namespace, STOPPED, time.Minute)
+
+	status, err := ContainerStatus(namespace)
+	assert.Nil(t, err)
+	assert.Equal(t, status, STOPPED)
+}
+
+func TestWaitForContainerRunning(t *testing.T) {
+	namespace := []string{"TestWaitForContainerRunning"}
+	startTestService(namespace)
+	defer StopService(namespace)
+	err := <-WaitContainerStatus(namespace, RUNNING, time.Minute)
+	assert.Nil(t, err)
+}
+
+func TestWaitForContainerTimeout(t *testing.T) {
+	namespace := []string{"TestWaitForContainerTimeout"}
+	startTestService(namespace)
+	defer StopService(namespace)
+	err := <-WaitContainerStatus(namespace, RUNNING, time.Nanosecond)
+	assert.NotNil(t, err)
+}
+
+func TestWaitForContainerStopped(t *testing.T) {
+	namespace := []string{"TestWaitForContainerStopped"}
+	startTestService(namespace)
+	<-WaitContainerStatus(namespace, RUNNING, time.Minute)
+
+	StopService(namespace)
+	err := <-WaitContainerStatus(namespace, STOPPED, time.Minute)
+	assert.Nil(t, err)
+}
+
+// WaitContainerStatus wait for the container to have the provided status until it reach the timeout
+func WaitContainerStatus(namespace []string, status StatusType, timeout time.Duration) (wait chan error) {
```

Comment:
This function is here for the test to wait enough time to pass. It should be useless in the future (when the stop and start wait for container).